internal/api/utils.go

```diff
@@ -308,6 +326,36 @@ func truncateComponentUnits(recommendationJSON map[string]interface{}) map[strin
 			log.Errorf("error truncating duration_in_hours in term %s\n", period)
 		}
 
+		if plotsObject, ok := intervalData["plots"].(map[string]interface{}); ok {
+			if plotsDataObject, ok := plotsObject["plots_data"].(map[string]interface{}); ok {
+				for _, value := range plotsDataObject {
+					if datapointMap, ok := value.(map[string]interface{}); ok {
+						if cpuUsage, ok := datapointMap["cpuUsage"].(map[string]interface{}); ok {
+							cpuUnit := unitsToTransform["cpu"]
+							if _, ok := cpuUsage["format"].(string); ok {
+								cpuUsage["format"] = cpuUnit
+							}
+							for _, key := range []string{"q1", "q3", "min", "max", "median"} {
+								cpuValue, _ := cpuUsage[key].(float64)
+								cpuUsage[key] = convertCPUUnit(cpuUnit, cpuValue)
+							}
+						}
+						if memoryUsage, ok := datapointMap["memoryUsage"].(map[string]interface{}); ok {
+							memoryUnit := unitsToTransform["memory"]
+							if _, ok := memoryUsage["format"].(string); ok {
+								memoryUsage["format"] = memoryUnit
+							}
+							for _, key := range []string{"q1", "q3", "min", "max", "median"} {
+								memoryValue, _ := memoryUsage[key].(float64)
+								memoryUsage[key] = convertMemoryUnit(memoryUnit, memoryValue)
+							}
+						}
```

Comment:
Future scope, not a blocker, maybe if there is performance challenge we can run cpu and memory conversations as go routine.

Reply:
Sure, we could look into it